src/MongoDB.Driver.Core/Core/Configuration/CryptClientSettings.cs

```diff
@@ -0,0 +1,122 @@
+﻿/* Copyright 2010-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Collections.Generic;
+using MongoDB.Bson;
+using MongoDB.Driver.Core.Encryption;
+
+namespace MongoDB.Driver.Core.Configuration
+{
+    /// <summary>
+    /// Represents settings for a crypt client.
+    /// </summary>
+    public sealed class CryptClientSettings
+    {
+        /// <summary>
+        /// Gets a value indicating whether query analysis should be bypassed.
+        /// </summary>
+        public bool? BypassQueryAnalysis { get; }
```

Comment:
I don't really know what to do with this design, previously we agreed that we can do it only for small classes with like few fields without logic inside, this case doesn't suit to this idea. I will rely on the team opinion here.

Reply:
we have discussed this question or retro. Some output:

Advantages of using background fields:
1. Sooner or later, properties will contain additional/validation logic that will require background field and it's better to not combine 2 of these approaches
2. Having a single place where you can find everything related to internal class state (background fields) and external class state (properties) is not a bad idea
3. 99% of our code is already written in a way with background fields, there is no significant need to change it even though we all may have different preferences in how we would expect writing code

Disadvantages:
4. We can save a bit code space in some cases if we use just auto properties

cc: @JamesKovacs , @rstam